File: bianry-watch/binary-watch.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;
class Solution {
public:
    vector<string> readBinaryWatch(int num) {
        vector<string> result;
        for (int h = 0; h < 12; ++h) {
            for (int m = 0; m < 60; ++m) {
                if (bit_count(h) + bit_count(m) == num) {
                    const auto hour = to_string(h);
                    const auto minute = m < 10 ? "0" + to_string(m) : to_string(m);
                    result.emplace_back(hour + ":" + minute);
                }
            }
        }
        return result;
    }

private:
    int bit_count(int bits) {
        int count = 0;
        for (; bits; bits &= bits - 1) {
            ++count;
        }
        return count;
    }
};
int main()
{
	Solution s;
	vector<string> time = s.readBinaryWatch(2);
	for (int i=0; i<time.size(); ++i)
	{
		cout<< time.at(i) <<endl;
	}
}

File: bianry-watch/binary_watch.py
class solution(object):
	def binary_watch(self,num):
		return["{0}:{1}".format(str(h), str(m).zfill(2))
		        for h in range(12) for m in range(60)
			if (bin(h)+bin(m)).count('1') == num]


if __name__ == '__main__':
	s = solution()
	result = s.binary_watch(2)
	print(result)	
